### PR TITLE
Change inv method to adjoint method in unit tests

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

`inv()` methods are removed from Python unit tests based on recent changes in [Pennylane](https://github.com/PennyLaneAI/pennylane/pull/3618) recently.

**Description of the Change:**

`inv()` methods are changed to `qml.adjoint()` methods Python unit tests.

**Benefits:**

This will avoid the CI failures caused by removing `inv()` methods from Pennylane.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: tests/test_gates.py [hunk 1]
```diff
@@ -138,51 +138,6 @@ def output(input):
     assert np.allclose(unitary, np.square(unitary_expected))
 
 
-@pytest.mark.parametrize("op_name", LightningGPU.operations)
-def test_inverse_unitary_correct(op, op_name):
```
Comment (mlxd) [inline tests/test_gates.py:142]: As mentioned earlier, let's update this to use the new way replacing `.inv()`

**Review by mlxd — CHANGES_REQUESTED** [review]
Thanks @multiphaseCFD 
I think we need to update, rather than remove these tests. See https://github.com/PennyLaneAI/pennylane/blob/master/doc/releases/changelog-dev.md#breaking-changes

File: tests/test_adjoint_jacobian.py [hunk 2]
```diff
@@ -639,23 +638,18 @@ def circuit_ansatz(params, wires):
     qml.QubitStateVector(unitary_group.rvs(2**4, random_state=0)[0], wires=wires)
     qml.RX(params[0], wires=wires[0])
     qml.RY(params[1], wires=wires[1])
-    qml.RX(params[2], wires=wires[2]).inv()
```
[the remark this answers is not in the document]
Comment (AmintorDusko) [reply tests/test_adjoint_jacobian.py:642]: [Some examples.](https://github.com/PennyLaneAI/pennylane-lightning/pull/365/files#diff-47426f63fedf47301eea9614251dad55036e7ddcb51a68605ee3975703926ef3)

[the remark this answers is not in the document]
Comment (multiphaseCFD) [reply tests/test_adjoint_jacobian.py:642]: > Same with all other tests removed.

Done!

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -6,14 +6,19 @@
 
 ### Improvements
 
+* Change `inv()` method to `qml.adjoint()` method in Python unit tests to be aligned with recent change in Pennylane and avoid CI failures.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:9]:
```suggestion
* Update `inv()` to `qml.adjoint()` in Python tests following recent changes in Pennylane.
```

File: tests/test_gates.py [hunk 4]
```diff
@@ -37,58 +37,56 @@
...
-        "SingleExcitationPlus": qml.SingleExcitationPlus(0.123, wires=[0, 3]),
-        "SingleExcitationMinus": qml.SingleExcitationMinus(0.123, wires=[0, 3]),
-        "DoubleExcitation": qml.DoubleExcitation(0.123, wires=[0, 1, 2, 3]),
-        "DoubleExcitationPlus": qml.DoubleExcitationPlus(0.123, wires=[0, 1, 2, 3]),
-        "DoubleExcitationMinus": qml.DoubleExcitationMinus(0.123, wires=[0, 1, 2, 3]),
-        "QFT": qml.QFT(wires=0),
-        "QubitSum": qml.QubitSum(wires=[0, 1, 2]),
-        "QubitCarry": qml.QubitCarry(wires=[0, 1, 2, 3]),
-        "QubitUnitary": qml.QubitUnitary(np.eye(2) * 1j, wires=0),
+        "RX": [qml.RX, [], {"phi": 0.123, "wires": [0]}],
```
Comment (AmintorDusko) [inline tests/test_gates.py:40]: Nice that you brought this update from lightning too! :slightly_smiling_face:

**Review by AmintorDusko — APPROVED** [review]
Nice work! Thank you! I just made one suggestion for the changelog, but nothing is blocking it from my side.

File: .github/CHANGELOG.md (hunk 3 above)
Comment (multiphaseCFD) [reply .github/CHANGELOG.md:9]: Thanks @AmintorDusko ! Let me update it.

File: tests/test_gates.py (hunk 4 above)
Comment (multiphaseCFD) [reply tests/test_gates.py:40]: Thanks for the nice reference @AmintorDusko !

**Review by mlxd — APPROVED** [review]

**Pull request merged**